### PR TITLE
update datadog api go client to 1.7.0

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Update datadog api go client to 1.7.0

### Motivation

Have the most up to date client that supports more recent DD features
### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Make sure the DatadogMonitor controller works as expected (create monitor, update, monitor, delete monitor)

### CODE REVIEW

File: controllers/datadogmonitor/monitor.go [hunk 1]
```diff
@@ -168,7 +166,11 @@ func buildMonitor(logger logr.Logger, dm *datadoghqv1alpha1.DatadogMonitor) (*da
 }
 
 func getMonitor(auth context.Context, client *datadogapiclientv1.APIClient, monitorID int) (datadogapiclientv1.Monitor, error) {
-	m, _, err := client.MonitorsApi.GetMonitor(auth, int64(monitorID)).GroupStates("all").Execute()
+	groupStates := "all"
```
Comment (clamoriniere) [inline controllers/datadogmonitor/monitor.go:169]: could be move as a const string

File: controllers/datadogmonitor/monitor.go [hunk 2]
```diff
@@ -209,7 +211,11 @@ func updateMonitor(auth context.Context, logger logr.Logger, client *datadogapic
 }
 
 func deleteMonitor(auth context.Context, client *datadogapiclientv1.APIClient, monitorID int) error {
-	if _, _, err := client.MonitorsApi.DeleteMonitor(auth, int64(monitorID)).Execute(); err != nil {
+	force := "false"
```
Comment (clamoriniere) [inline controllers/datadogmonitor/monitor.go:214]: could be move as a const string

**Review by clamoriniere — APPROVED** [review]
looks good, 2 small nits

**Comment by celenechang** [comment]
talked about the comments directly, decided to leave it be since we cannot get the address of constants

**Pull request merged**